h5py/_hl/base.py

```diff
@@ -418,3 +418,15 @@ def __eq__(self, other):
 
     def __repr__(self):
         return "Empty(dtype={0!r})".format(self.dtype)
+
+
+def product(nums):
```

Comment:
do we want to make this `_product` to protect it a bit from users?

Reply:
I figure that anything you have to import from `h5py._hl` is effectively private anyway - the public API is exposed on the top-level `h5py` module. But I'm happy to add a `_` if you prefer.